### ISSUE TITLE
Tensorboard SummaryWriter.add_hparams does not log hparam metric with spaces

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/pytorch#28765]
## To Reproduce
The following code logs correctly
```python
from torch.utils.tensorboard import SummaryWriter
writer = SummaryWriter(log_dir="/tmp/tb_logs")
writer.add_hparams({"a": 0, "b": 0}, {"hparam/test_accuracy": 0.5})
writer.add_hparams({"a": 0, "b": 1}, {"hparam/test_accuracy": 0.6})
writer.add_hparams({"a": 1, "b": 0}, {"hparam/test_accuracy": 0.61})
writer.add_hparams({"a": 1, "b": 1}, {"hparam/test_accuracy": 0.4})
writer.add_hparams({"a": 2.0, "b": 1.5}, {"hparam/test_accuracy": 0.7})
```

<img width="1436" alt="Screen Shot 2019-10-27 at 23 26 49" src="https://user-images.githubusercontent.com/2459423/67642957-8f924980-f911-11e9-81af-64bbacdda12f.png">

and this code does not log `hparam/test accuracy` values
```python
from torch.utils.tensorboard import SummaryWriter
writer = SummaryWriter(log_dir="/tmp/tb_logs")
writer.add_hparams({"a": 0, "b": 0}, {"hparam/test accuracy": 0.5})
writer.add_hparams({"a": 0, "b": 1}, {"hparam/test accuracy": 0.6})
writer.add_hparams({"a": 1, "b": 0}, {"hparam/test accuracy": 0.61})
writer.add_hparams({"a": 1, "b": 1}, {"hparam/test accuracy": 0.4})
writer.add_hparams({"a": 2.0, "b": 1.5}, {"hparam/test accuracy": 0.7})
```
<img width="1433" alt="Screen Shot 2019-10-27 at 23 28 25" src="https://user-images.githubusercontent.com/2459423/67642958-8f924980-f911-11e9-9caa-b4831c7dc853.png">

## Expected behavior

Normally, metric names with spaces are preprocessed : space is replaced by `_` .

## Environment

Collecting environment information...
PyTorch version: 1.3.0
Is debug build: No
CUDA used to build PyTorch: 10.1.243

OS: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.10.2

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 430.26
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.3

Versions of relevant libraries:
[pip3] botorch==0.1.4
[pip3] gpytorch==0.3.6
[pip3] numpy==1.17.2
[pip3] pytorch-ignite==0.3.0
[pip3] torch==1.3.0
[pip3] torchfile==0.1.0
[pip3] torchvision==0.4.1
[conda] blas                      1.0                         mkl  
[conda] botorch                   0.1.4                    pypi_0    pypi
[conda] gpytorch                  0.3.6                    pypi_0    pypi
[conda] mkl                       2019.4                      243  
[conda] mkl-include               2019.4                      243  
[conda] mkl-service               2.3.0            py37he904b0f_0  
[conda] mkl_fft                   1.0.14           py37ha843d7b_0  
[conda] mkl_random                1.1.0            py37hd6b4f25_0  
[conda] pytorch                   1.3.0           py3.7_cuda10.1.243_cudnn7.6.3_0    pytorch
[conda] pytorch-ignite            0.3.0                    pypi_0    pypi
[conda] torchfile                 0.1.0                    pypi_0    pypi
[conda] torchvision               0.4.1                    pypi_0    pypi

**Comment by natalialunova** [comment]
Hi @vfdev-5! Thank you for reporting this issue, but we decided do not replace whitespaces with `_`. E. g. if user has "key 1" and "key_1", the replacing logic would lead to dictionary key corruption.

**Comment by lanpa** [comment]
Agreed, also the metric value is usually from a torch tensor. A key with the same name of its corresponding tensor name makes the code cleaner. (variable name cannot contain spaces)
